docs/developers/developing_iree/testing_guide.md

```diff
@@ -3,6 +3,95 @@
 Like the IREE project in general, IREE tests are divided into a few different
 components and use different tooling depending on the needs of that component.
 
+| Test type       | Test                                            | Build system | Supported platforms |
+|:--------------  | :-----------------                              | -----------  | -------------       |
+| Compiler tests  | iree_lit_test                                   | Bazel/CMake  | Host                |
+| Runtime tests   | iree_cc_test                                    | Bazel/CMake  | Host/Device         |
+|                 | iree_native_test                                | Bazel/CMake  | Host/Device         |
+|                 | iree_hal_cts_test_suite                         | CMake        | Host/Device         |
+| Core E2E tests  | iree_check_test                                 | Bazel/CMake  | Host/Device         |
+|                 | iree_trace_runner_test                          | Bazel/CMake  | Host/Device         |
+|                 | iree_single_backend_generated_trace_runner_test | Bazel/CMake  | Host/Device         |
+|                 | iree_generated_trace_runner_test                | Bazel/CMake  | Host/Device         |
+|                 | iree_static_linker_test                         | CMake        | Host                |
```

Comment:
Remove `iree_run_module_test` and the `Host/Device` support for `iree_static_linker_test`. I will update the table after the corresponding test functions actually landed.